server/src/main/java/io/druid/segment/realtime/plumber/RealtimePlumber.java

```diff
@@ -417,7 +459,7 @@ public void doRun()
               synchronized (hydrant) {
                 if (!hydrant.hasSwapped()) {
                   log.info("Hydrant[%s] hasn't swapped yet, swapping. Sink[%s]", hydrant, sink);
-                  final int rowCount = persistHydrant(hydrant, schema, interval);
+                  final int rowCount = persistHydrant(hydrant, schema, interval, null);
```

Comment:
Ok, do you want to worry about making things actually work for this PR, or just get it merged and change things later? Fwiw I'm ok with the second. I think a good fix would be to move the interface line a bit such that the persisting and pushing is handled by something different than what handles the indexing and serving, which is probably a bigger change than should go in this PR